src/docs/perf/deferred-components.md

```diff
@@ -0,0 +1,589 @@
+
+To support `SplitCompat`, there are three methods
+(as detailed in the [Android docs][]), any of which are valid:
+
+* Make your application class extend SplitCompatApplication:
+  <!-- skip -->
+  ```dart
+  public class MyApplication extends SplitCompatApplication {
+      ...
+  }
+  ```
+
+* Call `SplitCompat.install(this);` in `attachBaseContext`:
+  <!-- skip -->
+  ```dart
+  @Override
+  protected void attachBaseContext(Context base) {
+      super.attachBaseContext(base);
+      // Emulates installation of future on demand modules using SplitCompat.
+      SplitCompat.install(this);
+  }
+      ```
+* Declare `SplitCompatApplication` as the application
+  subclass and add the flutter compatibility code from
+  `FlutterApplication` to your application class:
+  ```js
+  <application
+      ...
+      android:name="com.google.android.play.core.splitcompat.SplitCompatApplication">
+  </application>
+  ```
+  The embedder relies on an injected
+  `DeferredComponentManager` instance to handle
+  install requests for deferred components.
+  Provide a `PlayStoreDeferredComponentManager` into
+  the Flutter embedder by adding the following code
+  to your app initialization:
+  <!-- skip -->
+  ```dart
+  import io.flutter.embedding.engine.dynamicfeatures.PlayStoreDeferredComponentManager;
+  import io.flutter.FlutterInjector;
+  ... 
+  PlayStoreDeferredComponentManager deferredComponentManager = new
+      PlayStoreDeferredComponentManager(this, null);
+  FlutterInjector.setInstance(new FlutterInjector.Builder()
+      .setDeferredComponentManager(deferredComponentManager).build());
+  ```
+
+</li>
+    
+<li markdown="1">Opt into deferred components by adding
+    the `deferred-components` entry to the app’s `pubspec.yaml`
+    under the `flutter` entry:
+
+  ```yaml
+  ...
+  flutter:
+    ...
+    deferred-components:
+    ...
+  ```
+  The `flutter` tool looks for the `deferred-components`
+  entry in the `pubspec.yaml` to determine whether the
+  app should be built as deferred or not.
+  This can be left empty for now unless you already
+  know the components desired and the Dart deferred libraries
+  that go into each. You will fill in this section later
+  in [step 3.3][] once `gen_snapshot` produces the loading units.
+</li>
+</ol>
+
+### Step 2: Implementing deferred Dart libraries
+
+Next, use deferred loaded Dart libraries in your
+app’s Dart code. The implementation does not need
+to be feature complete yet. The example in the
+rest of this page adds a new simple deferred widget
+as a placeholder. You can also convert existing code
+to be deferred by modifying the imports and
+guarding usages of deferred code behind [`loadLibrary()`][]
+`Futures`.
+
+<ol markdown="1">
+<li markdown="1">Create a new Dart library.
+    For example, create a new `DeferredBox` widget that
+    can be downloaded at runtime.
+    This widget can be of any complexity but,
+    for now, create a simple box as a stand-in.
```

Comment:
for now -> for the purposes of this guide, we will use a simple box as a stand-in